### PR TITLE
mgmt network, applicationgateway support create default with BASIC tier and sku

### PR DESCRIPTION
**Pull request by XiaofeiCao**
# Description

issue: https://github.com/Azure/azure-sdk-for-java/issues/36215

When user doesn't specify `tier` and `sku` during creation, will use `BASIC` as default, as opposed to `STANDARD_SMALL` which no longer works.

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-network/src/test/java/com/azure/resourcemanager/network/ApplicationGatewayTests.java [hunk 1]
```diff
@@ -682,6 +682,46 @@ public void canSetSslPolicy() {
...
+
+        String appPublicIp = generateRandomResourceName("pip", 15);
+
+        PublicIpAddress pip =
+            networkManager
+                .publicIpAddresses()
+                .define(appPublicIp)
+                .withRegion(Region.US_EAST)
+                .withNewResourceGroup(rgName)
+                .withSku(PublicIPSkuType.STANDARD)
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-network/src/test/java/com/azure/resourcemanager/network/ApplicationGatewayTests.java:697]: Does AppGateway work with BASIC ipAddress?

Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-network/src/test/java/com/azure/resourcemanager/network/ApplicationGatewayTests.java:697]: No it doesn't... It's more like a V2 SKU to me.
> Application Gateway with SKU Basic can only reference public ip with Standard SKU

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-network/src/test/java/com/azure/resourcemanager/network/ApplicationGatewayTests.java:697]: Got it. Do you know whether `withNewPublicIpAddress()` creates an IP address?

**Review by weidongxu-microsoft — APPROVED** [review]

File: sdk/resourcemanager/azure-resourcemanager-network/CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
 - Supported `backends` for `LoadBalancingRule`.
 - Supported `withSslPolicy` for `ApplicationGateway`.
+- Now `ApplicationGateway` can be created with `BASIC` sku and tier if not specified.
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-network/CHANGELOG.md:9]: Should we move to breaking changes for notification for behavior change.

```
No new V1 deployments: August 28, 2024 - V1 creation is stopped completely for all customers 28 August 2024 onwards.
```

Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-network/CHANGELOG.md:9]: Sure.

File: sdk/resourcemanager/azure-resourcemanager-network/src/test/java/com/azure/resourcemanager/network/ApplicationGatewayTests.java (hunk 1 above)
Comment (XiaofeiCao) [reply sdk/resourcemanager/azure-resourcemanager-network/src/test/java/com/azure/resourcemanager/network/ApplicationGatewayTests.java:697]: Yeah, it'll be created in `createInner`, through `ensureDefaultPipDefinition`:
https://github.com/Azure/azure-sdk-for-java/blob/654650746b7294af2f81457cde7db9afbbee76e2/sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/implementation/ApplicationGatewayImpl.java#L563-L573

Maybe we'll set pip sku to `STANDARD` here if it's not legacy gateway.

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-network/src/test/java/com/azure/resourcemanager/network/ApplicationGatewayTests.java:697]: Maybe next year. As currently user can still create V1 (if subscription has V1)

**Pull request merged**